test: upstream support for provider delayed shutdown (#5709)

steinbro opened the pull request:
- [ ] Have you followed the [guidelines for contributing](https://github.com/canonical/snapcraft/blob/main/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint`?
- [ ] Have you successfully run `make test`?

---
@lengau and @mr-cal, since this feature is already being pulled into craft-application via https://github.com/canonical/craft-application/pull/805, is there any need to do anything here beyond bumping the craft-application dependency version? Or was the point of #5709 to do just that and just check that nothing broke?

mr-cal reviewed (commented):
I think this typing of demo gives us the confidence to land https://github.com/canonical/craft-application/pull/805, although I'd set the variable in the top level `spread.yaml` and add the "run manual spread" label so all spread tests will run with it set.

steinbro commented:
> I think this typing of demo gives us the confidence to land [canonical/craft-application#805](https://github.com/canonical/craft-application/pull/805), although I'd set the variable in the top level `spread.yaml` and add the "run manual spread" label so all spread tests will run with it set.

I moved the variable to the top level spread configuration, but I don't know what you mean by "run manual spread" label.

mr-cal commented:
Re: https://github.com/canonical/snapcraft/actions/runs/17243808006/job/48928070604?pr=5725#step:4:2226

You'll need to apply these changes: https://github.com/canonical/snapcraft/pull/5349/commits/215b1d372565ba44f98219a9e5bec896799104ec

mr-cal commented:
And because of how the labels work, you have to remove and re-add the manual-spread label every time you push a new commit to re-trigger the manual spread tests

steinbro commented:
Per discussion with @mr-cal, the failing spread tests appeared to be due to the craft-application branch being too old and not having more recent fixes. I rebased https://github.com/canonical/craft-application/pull/805 as commit https://github.com/canonical/craft-application/commit/61aec610c6c9ea74a1cd5d130ac4217439831d9f and updated the dependency here accordingly.

mr-cal commented:
> @mr-cal can you confirm if the failing spread tests are acceptable? I'm not seeing the state manager anymore, just errors from apt refresh and the store.

@steinbro, agreed these all look like network-related failures.

lengau commented on requirements-noble.txt:
Looks like this needs reverting

steinbro commented:
@lengau do you know what's going on with the unit tests in CI now?

steinbro commented:
@lengau so this is mergeable now, once @bepri approves?
